### ISSUE TITLE
Missing keyword parameters for `__init_subclass__` of an ABC not caught

### ISSUE THREAD

**Issue by fracsinus** [microsoft/pyright#7819]
**Describe the bug**
Missing keyword parameters for `__init_subclass__` is not reported if the class inherits `abc.ABC`

**Code or Screenshots**
```python
from abc import ABC


class Foo(ABC):
    def __init_subclass__(cls, x): ...


class Bar:  # without ABC for comparison
    def __init_subclass__(cls, x): ...


class FooChild(Foo): ...  # this line is not caught by pyright and raises in runtime 


class BarChild(Bar): ...  # Incorrect keyword arguments for __init_subclass__ method
```

**VS Code extension or command-line**
pyright-langserver --stdio (1.1.361)

**Comment by erictraut** [comment]
If a class is created from a metaclass that overrides the `type.__new__` method, pyright will check that all keyword arguments passed in the `class` statement are accepted by the `__new__` method. It doesn't check the `__init_subclass__` in this case because it can't assume that the custom `__new__` method is going to call `__init_subclass__`. 

The `ABCMeta` metaclass overrides the `__new__` method of `type`, so it falls in this category.

I'll look into special-casing `ABCMeta` since we know it calls through to `type.__new__`. Unfortunately, it looks like this might create other problems because typeshed defines `_TypedDict` as using the `ABCMeta` metaclass, but it doesn't override `object.__init_subclass__`. That means all `TypedDict` classes that use `total` or `closed` keyword arguments in their class definition will generate type errors.

I'll give this a bit more thought, but I may end up closing this as "won't fix".

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.362, which I just published.
